### PR TITLE
Fixed multiple bugs with RTFM

### PR DESCRIPTION
**Pull request by Coler6gamer**
- Fixed a bug from thommo's improvements.
- More relevant matches will go to the top of autocomplete/embed.
- Fixed links not being specific for functions.
- Fixed a few other kinks.

### CODE REVIEW

File: carberretta/extensions/rtfm.py [hunk 1]
```diff
@@ -79,11 +88,12 @@ async def build_rtfm_output(query: str, cache: CachedObjT) -> hikari.Embed:
...
-            )
-
+            if cache[match][1][-1:] == "$":
+                description.append(
+                    f"[`{match}`]({url}{cache[match][1][:-1] + cache[match][0][0]})"
+                )
+            else:
+                description.append(f"[`{match}`]({url}{cache[match][1]})")
     embed.description = "\n".join(description)
     return embed
```
Comment (mshaugh) [inline carberretta/extensions/rtfm.py:98]: Compute the description first and then return the embed.

```python
matches = await get_rtfm(query, cache)
description = []
for match in matches:
    ...

return hikari.Embed(
    title: "RTFM",
    description: "\n".join(description),
    ...
)
```

File: carberretta/extensions/rtfm.py [hunk 2]
```diff
@@ -64,10 +66,17 @@ async def on_started(_: hikari.StartedEvent) -> None:
...
     matches = process.extract(value, cache.keys(), scorer=fuzz.QRatio, limit=15)
-    return [result for result, _, _ in matches]
+    match = []
+    pure_matches = []
+    for result, _, _ in matches:
+        if value in result:
+            pure_matches.append(result)
+        else:
+            match.append(result)
+    return pure_matches + match
```
Comment (mshaugh) [inline carberretta/extensions/rtfm.py:76]: Change the variable names.
I suggest: `matches` -> `extracted`, `match` -> `matches`.

File: carberretta/extensions/rtfm.py [hunk 3]
```diff
@@ -79,11 +88,12 @@ async def build_rtfm_output(query: str, cache: CachedObjT) -> hikari.Embed:
...
-                f"[`{match}`]({HIKARI_DOCS_URL}"
-                f"{plugin.bot.d.hikari_cache[match][1]})\n"
-            )
-
+            if cache[match][1][-1:] == "$":
+                description.append(
+                    f"[`{match}`]({url}{cache[match][1][:-1] + cache[match][0][0]})"
+                )
+            else:
+                description.append(f"[`{match}`]({url}{cache[match][1]})")
```
Comment (mshaugh) [inline carberretta/extensions/rtfm.py:96]: Improve readability with a `namedtuple` for cache entries.

**Review by mshaugh — DISMISSED** [review]

**Comment by Coler6gamer** [comment]
@parafoxia review please :)

File: carberretta/extensions/rtfm.py [hunk 4]
```diff
@@ -193,12 +214,10 @@ def decompress(bytes_obj: io.BytesIO) -> t.Generator[bytes, None, None]:
         if not (match := CHUNK_REGEX.match(line.rstrip())):
             continue
 
-        if match in cache:
+        direct, type, _, link, _ = match.groups()
+        if direct in cache.keys():
```
Comment (parafoxia) [inline carberretta/extensions/rtfm.py:218]:
```suggestion
        if direct in cache:
```

**Review by parafoxia — CHANGES_REQUESTED** [review]
Just a small change, but other than that it looks fine to me. Would be worth @mshaugh re-reviewing this to make sure all his requested changes are sufficient in his eyes.

**Review by parafoxia — APPROVED** [review]

**Comment by parafoxia** [comment]
@mshaugh, you still need to review this.

**Pull request merged**